MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SymbolLayerMapillaryActivity.java

```diff
@@ -514,20 +513,28 @@ private void toggleFavourite(int index) {
     ImageView imageView = view.findViewById(R.id.logoView);
     imageView.setImageResource(currentState ? R.drawable.ic_favorite : R.drawable.ic_favorite_border);
     Bitmap bitmap = SymbolGenerator.generate(view);
-    style.addImage(title, bitmap);
-    refreshSource();
+    mapboxMap.getStyle(new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        style.addImage(title, bitmap);
+        refreshSource();
+      }
+    });
   }
 
   /**
    * Invoked when the bitmaps have been generated from a view.
    */
   public void setImageGenResults(HashMap<String, View> viewMap, HashMap<String, Bitmap> imageMap) {
-    if (style.isFullyLoaded()) {
-      // calling addImages is faster as separate addImage calls for each bitmap.
-      style.addImages(imageMap);
-    }
-    // need to store reference to views to be able to use them as hitboxes for click events.
-    this.viewMap = viewMap;
+    mapboxMap.getStyle(new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        // calling addImages is faster as separate addImage calls for each bitmap.
+        style.addImages(imageMap);
+        // need to store reference to views to be able to use them as hitboxes for click events.
+        SymbolLayerMapillaryActivity.this.viewMap = viewMap;
```

Comment:
Shall we move this line outside the `getStyle` callback?

Reply:
Sure. Moved.